src/declaration.c

```diff
@@ -1653,6 +1653,7 @@ void VarDeclaration::setFieldOffset(AggregateDeclaration *ad, unsigned *poffset,
             {
                 const char *s = (t->ty == Tsarray) ? "static array of " : "";
                 ad->error("cannot have field %s with %ssame struct type", toChars(), s);
+                return;
```

Comment:
When a recursive struct field found, ad->sizeof should not be set to `SIZEOKfwd`, because it will re-run semantic to resolve forward reference outside of the gagging scope.